src/main/java/net/mcreator/io/zip/ZipIO.java

```diff
@@ -36,13 +36,12 @@
 
 	private static final Logger LOG = LogManager.getLogger("ZipIO");
 
-	public static void iterateZip(File zipFilePointer, IZipFileAction action) {
+	public static void iterateZip(File zipFilePointer, Consumer<ZipEntry> action, boolean sortByName) {
 		try (ZipFile zipFile = new ZipFile(zipFilePointer)) {
-			Enumeration<? extends ZipEntry> entries = zipFile.entries();
-			while (entries.hasMoreElements()) {
-				ZipEntry zipEntry = entries.nextElement();
-				action.process(zipEntry);
-			}
+			List<? extends ZipEntry> entries = Collections.list(zipFile.entries());
+			if (sortByName)
+				entries.sort(Comparator.comparing(e -> e.getName().toLowerCase()));

```

Comment:
I wouldn't lowercase here, as this files with uppercase should have priority before lowercase, this way, you can also use method reference e::getName

Reply:
Why is this priority ignored by Windows?

Reply:
Windows is case insensitive os, MCreator is case sensitive app